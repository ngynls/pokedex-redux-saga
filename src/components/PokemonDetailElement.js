import React, { Component } from "react";
import PokedexUtils from "../components/shared/PokedexUtils";

class PokemonDetailElement extends Component {

    displayTypes= () => {
        return this.props.types.map((type, i) => {
            let typeColor = PokedexUtils.getTypeColor(type.type.name);
            return (
                <span 
                className="badge badge-pill badge-primary" 
                key={ i }
                style={ { backgroundColor: typeColor } }
                >
                        { PokedexUtils.capitalizeName(type.type.name) }
                </span>
            );
        });
    }
    
    displayStats= () => {
        return this.props.stats.map((stat, i) => {
            return (
                <div className="stat-element"  key={ i }>
                    <span><b>{ PokedexUtils.capitalizeName(stat.stat.name) }</b> : { stat.base_stat }</span>
                    <br/>
                </div>
            );
        });
    }

    displayAbilities= () => {
        return this.props.abilities.map((ability, i) => {
            return (
                <div className="ability-element" key={ i }>
                <span>{ PokedexUtils.capitalizeName(ability.ability.name) }</span>
                </div>
            );
        });
    }

    render() {
        const { id, name, types, height, weight, description, spriteUrl, stats, abilities } = this.props;
            return (
                <div className="details-container">
                    <div className="jumbotron">
                        <img src={ spriteUrl } alt="sprite"/>
                    </div>
                    <div className="card">
                        <div className="card-body">
                            <div className="name-section"><h2>{ name }</h2></div>
                            <div className="id-section"><b>Id:</b> #{ id }</div>
                            <div className="type-section"><b>Type: </b>{
                                types ? this.displayTypes() : null
                            }
                            </div>
                            <div className="height-section"><b>Height: </b>{ height } m</div>
                            <div className="weight-section"><b>Weight: </b>{ weight } kg</div>
                            <div className="description-section"><b>Description: </b>{ description }</div>
                            <br/>
                            <div className="stats-section">
                                <h3>Base Stats</h3>
                                <hr></hr>
                                {
                                    stats ? this.displayStats() : null
                                }
                            </div>
                            <br/>
                            <div className="abilities-section">
                                <h3>Abilities</h3>
                                <hr></hr>
                                {
                                    abilities ? this.displayAbilities() : null
                                }
                            </div>
                        </div>
                    </div>
                </div>
            );
    }
}


export default PokemonDetailElement;