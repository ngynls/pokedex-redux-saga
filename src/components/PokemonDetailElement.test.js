import React from "react";
import renderer from "react-test-renderer";
import PokemonDetailElement from "./PokemonDetailElement";

test("matches snapshot of PokedexDetailElement", () => {
    const id = 1;
    const name = "Bulbasaur";
    const types = [
        {type: {name: "Poison"}},
        {type: {name: "Grass"}}
    ];
    const stats = [
        {
            base_stat: 45,
            stat:{name: "Speed"}
        },
        {
            base_stat: 65,
            stat:{name: "Special-defense"}
        },
        {
            base_stat: 65,
            stat:{name: "Special-attack"}
        },
        {
            base_stat: 49,
            stat:{name: "Defense"}
        },
        {
            base_stat: 49,
            stat:{name: "Attack"}
        },
        {
            base_stat: 45,
            stat:{name: "Hp"}
        }
    ];
    const abilities = [
        {ability: {name: "Chlorophyll"}},
        {ability: {name: "Overgrow"}}
    ];
    const height = 0.7;
    const weight = 6.9;
    const description = "Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sunâ€™s rays, the seed grows progressively larger.";
    const spriteUrl = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png";
    const tree = renderer.create(<PokemonDetailElement 
        id={id} 
        name={name} 
        types={types}
        stats={stats}
        abilities={abilities} 
        height={height} 
        weight={weight} 
        description={description}
        spriteUrl={spriteUrl}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});