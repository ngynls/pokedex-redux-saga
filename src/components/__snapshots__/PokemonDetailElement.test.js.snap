// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of PokedexDetailElement 1`] = `
<div
  className="details-container"
>
  <div
    className="jumbotron"
  >
    <img
      alt="sprite"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
  </div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <div
        className="name-section"
      >
        <h2>
          Bulbasaur
        </h2>
      </div>
      <div
        className="id-section"
      >
        <b>
          Id:
        </b>
         #
        1
      </div>
      <div
        className="type-section"
      >
        <b>
          Type: 
        </b>
        <span
          className="badge badge-pill badge-primary"
          style={
            Object {
              "backgroundColor": "#c5c5c5",
            }
          }
        >
          Poison
        </span>
        <span
          className="badge badge-pill badge-primary"
          style={
            Object {
              "backgroundColor": "#c5c5c5",
            }
          }
        >
          Grass
        </span>
      </div>
      <div
        className="height-section"
      >
        <b>
          Height: 
        </b>
        0.7
         m
      </div>
      <div
        className="weight-section"
      >
        <b>
          Weight: 
        </b>
        6.9
         kg
      </div>
      <div
        className="description-section"
      >
        <b>
          Description: 
        </b>
        Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sunâ€™s rays, the seed grows progressively larger.
      </div>
      <br />
      <div
        className="stats-section"
      >
        <h3>
          Base Stats
        </h3>
        <hr />
        <div
          className="stat-element"
        >
          <span>
            <b>
              Speed
            </b>
             : 
            45
          </span>
          <br />
        </div>
        <div
          className="stat-element"
        >
          <span>
            <b>
              Special-defense
            </b>
             : 
            65
          </span>
          <br />
        </div>
        <div
          className="stat-element"
        >
          <span>
            <b>
              Special-attack
            </b>
             : 
            65
          </span>
          <br />
        </div>
        <div
          className="stat-element"
        >
          <span>
            <b>
              Defense
            </b>
             : 
            49
          </span>
          <br />
        </div>
        <div
          className="stat-element"
        >
          <span>
            <b>
              Attack
            </b>
             : 
            49
          </span>
          <br />
        </div>
        <div
          className="stat-element"
        >
          <span>
            <b>
              Hp
            </b>
             : 
            45
          </span>
          <br />
        </div>
      </div>
      <br />
      <div
        className="abilities-section"
      >
        <h3>
          Abilities
        </h3>
        <hr />
        <div
          className="ability-element"
        >
          <span>
            Chlorophyll
          </span>
        </div>
        <div
          className="ability-element"
        >
          <span>
            Overgrow
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
